Enable all default golangci-lint linters and fix issues related

Jdubrick opened the pull request:
## What does this PR do?
This PR enables the default linters that were disabled in a prior issue and fixes all of the issues presented by the default linters.
<!-- _Summarize the changes_ -->

### Which issue(s) does this PR fix
resolves https://github.com/devfile/api/issues/1302
<!-- _Link to github issue(s)_ -->

### PR acceptance criteria

- [ ] Enable default linters
- [ ] All issues reported by `make lint` are dealt with and `make lint` passes
- [ ] All tests pass in `make test`

### How to test changes / Special notes to the reviewer
You can test these changes by first running `make lint` to verify there are no more lint errors being reported. You can also run `make test` to ensure all unit tests also pass with the changes.

thepetk reviewed (commented):
In general it looks good to me :) It would be nice if we could add some test cases for the coverage report that is failing.

Jdubrick commented:
> In general it looks good to me :) It would be nice if we could add some test cases for the coverage report that is failing.

Sounds good. Currently working on trying to increase code coverage for these failing portions, just have to familiarize myself with Go testing

thepetk commented:
> Pushed some changes that increase coverage a little bit for the defer functions and some error cases in `detector.go`, still doesn't loook like it hits the same coverage as before but it's only different by 0.03% from [this link](https://app.codecov.io/gh/devfile/alizer/pull/37?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile). What are your thoughts @thepetk ?

Could you try to add some test cases for the `DownloadDevfileTypesFromRegistry` too?

thepetk commented on pkg/apis/recognizer/devfile_recognizer.go:
```diff
@@ -268,6 +264,12 @@ var DownloadDevfileTypesFromRegistry = func(url string, filter model.DevfileFilt
 	return devfileTypes, nil
 }
 
+func closeHttpResponseBody(resp *http.Response){
```
@Jdubrick 

I'd suggest renaming this to `CloseHttpResponseBody` and place it inside the `detector.go` (utils package). This way it can be imported both in `recognizer` & `main` (check_registry) and we can have the implementation only once.

Initialy I thought we couldn't do that but it was okay when I tested it WDYT?

Jdubrick replied:
I will make the change!

Jdubrick commented on pkg/apis/recognizer/devfile_recognizer_test.go:
```diff
@@ -791,3 +857,31 @@ func getTestProjectPath(folder string) string {
+			if err != nil {
+				t.Errorf("Failed to get url")
+			}else {
+				closeHttpResponseBody(resp)
```
Good catch, I think that's a good idea. Initially I implemented it this way because I didn't know to to do the http test properly but we got it now. I'll make that change

Jdubrick commented:
> @thepetk I added more tests and was able to increase the patch code coverage as a result. From what I was reading online it seemed that a majority of people ignore the err that could arise from `resp.Body.Close()` as there isn't really anything you can do about it. In this case I can try and remove the err handling for the close, but then we get a potential problem warning from github, see above for reference to that. Once I reintroduced the error handling we dropped just below the 70% cutoff. What do you think should be the course of action?

Last commit fixes our issue as I was able to mock the function throwing an error.

thepetk commented:
> > @thepetk I added more tests and was able to increase the patch code coverage as a result. From what I was reading online it seemed that a majority of people ignore the err that could arise from `resp.Body.Close()` as there isn't really anything you can do about it. In this case I can try and remove the err handling for the close, but then we get a potential problem warning from github, see above for reference to that. Once I reintroduced the error handling we dropped just below the 70% cutoff. What do you think should be the course of action?
> 
> Last commit fixes our issue as I was able to mock the function throwing an error.

Very complicated failure from the codecov I have to say. I think the addition we make here with the separate functions will let us keep the coverage higher in the future, so is definitely a good addition.

thepetk reviewed (approved):
/lgtm
